pkg/utils/openshift/utils.go

```diff
@@ -15,10 +15,23 @@ func IsOpenShift(cfg *rest.Config) (bool, error) {
 
 /*
 GetPlatformInfo examines the Kubernetes-based environment and determines the running platform, version, & OS.
+Returned PlatformInfo.OCPVersion is derived from underlying PlatformInfo.K8SVersion.
 Accepts <nil> or instantiated 'cfg' rest config parameter.
 
-Result: PlatformInfo{ Name: OpenShift, OCPVersion: 4.1.0, K8SVersion: 1.13+, OS: linux/amd64 }
+Result: PlatformInfo{ Name: OpenShift, OCPVersion: 4.1, K8SVersion: 1.13+, OS: linux/amd64 }
 */
 func GetPlatformInfo(cfg *rest.Config) (platform.PlatformInfo, error) {
 	return platform.K8SBasedPlatformVersioner{}.GetPlatformInfo(nil, cfg)
 }
+
+/*
+LookupOpenShiftVersion refines PlatformInfo.OCPVersion info based on API calls, rather than deriving from K8SVersion.
+*** NOTE: OCP 4.1+ requires elevated user permissions, see PlatformVersioner for details
+Accepts <nil> or instantiated 'cfg' rest config parameter.
+Accepts prepopulated or empty PlatformInfo{}
+
+Result: PlatformInfo{ Name: OpenShift, OCPVersion: 4.1.2, K8SVersion: 1.13+, OS: linux/amd64 }
+*/
+func LookupOpenShiftVersion(cfg *rest.Config, info platform.PlatformInfo) (platform.PlatformInfo, error) {
```

Comment:
If I understand correctly, this is generally bad practice, in that you already have the PlatformInfo but is OCP version is not populated.